File: RestaurantApp/RestaurantApp/RestaurantApp/Models/Review.swift
//
//  Review.swift
//  RestaurantApp
//
//  Created by Азат Зиганшин on 07.05.2024.
//

import Foundation

struct Review: Codable {
    let id: Int
    let name: String
    let title: String
    let body: String
    let rating: Int
}

File: RestaurantApp/RestaurantAppServer/Domain/Entities/Review.cs
namespace Domain.Entities;

using System.ComponentModel.DataAnnotations;

public class Review
{
    [Key]
    public int Id { get; set; }
    
    [Required(ErrorMessage = "Name is required")]
    public string Name { get; set; }

    [Required(ErrorMessage = "Title is required")]
    public string Title { get; set; }

    [Required(ErrorMessage = "Review text is required")]
    public string Body { get; set; }

    [Range(1, 5, ErrorMessage = "Rating must be between 1 and 5")]
    public int Rating { get; set; }
}
